app/views/layouts/application.html.erb

```diff
@@ -40,7 +40,9 @@
         <% end %>
         <% if user_signed_in? %>
     			<div class="buttons">
-  	  			<button class="button"><%= link_to "New Post", new_post_path %></button>
+            <% if (@posts && policy(@posts).edit?) || (@post && policy(@post).edit?) %>
+    	  			<button class="button"><%= link_to "New Post", new_post_path %></button>
+            <% end %>
```

Comment:
This is tested and working as I want. This conditional was required to cover two scenarios: on the index page where there is a post collection being listed, and on the post/show page where a single member is listed. Let me know your thoughts.

Reply:
Got it, would be better to split into 2 separate views in future